packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt

```diff
@@ -158,23 +159,29 @@ class AppContext(
         val reactContext = reactContextHolder.get() ?: return@trace
         val jsContextHolder = reactContext.javaScriptContextHolder?.get() ?: return@trace
 
-        val jsCallInvokerHolder = reactContext.catalystInstance?.jsCallInvokerHolder
-        if (jsCallInvokerHolder !is CallInvokerHolderImpl) {
-          logger.warn("⚠️ Cannot install JSI interop: CallInvokerHolderImpl is not available")
+        val jsRuntimePointer = jsContextHolder.takeIf { it != 0L }.ifNull {
+          logger.error("❌ Cannot install JSI interop - JS runtime pointer is null")
           return@trace
         }
 
-        jsContextHolder
-          .takeIf { it != 0L }
-          ?.let {
-            jsiInterop.installJSI(
-              this,
-              it,
-              jniDeallocator,
-              jsCallInvokerHolder
-            )
-            logger.info("✅ JSI interop was installed")
-          }
+        @Suppress("DEPRECATION")
+        if (reactContext.isBridgeless) {
```

Comment:
nit: the method will likely be removed in the future. would be good to figure out a way with meta to check bridgeless mode from java side.

Reply:
I think we will replace that check with `is new arch enabled` in the future. Right now, we can't do it because someone could turn off the bridgeless but still have fabric enabled.

Reply:
alternatively to check `ReactApplication.reactNativeHost == null` and `ReactApplication.reactHost == null`
https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactApplication.kt

well, let's do that after the `isBridgeless` being removed.